distributed/tests/test_worker.py

```diff
@@ -3502,3 +3505,43 @@ async def test_reconnect_argument_deprecated(s):
         warnings.simplefilter("error")
         async with Worker(s.address):
             pass
+
+
+@gen_test()
+async def test_worker_makes_own_thread():
+    class InitWorkerNewThread(WorkerPlugin):
+        name: name
+        thread: threading.Thread
+
+        def __init__(self, scheduler):
+            super().__init__()
+            self.scheduler = scheduler
+
+        def setup(self, worker):
+            self.loop = asyncio.new_event_loop()
+            self.thread = threading.Thread(target=self.loop.run_forever)
+            self.thread.start()
+            future = asyncio.run_coroutine_threadsafe(self.func(), self.loop)
+            future.result()
+
+        def teardown(self, worker):
+            self.loop.call_soon_threadsafe(self.loop.stop)
+            self.thread.join(timeout=10)
+
+        async def func(self):
+            client = await Client(self.scheduler, asynchronous=True)
+            try:
+                get_worker()
+            except ValueError:
+                await client.register_worker_plugin(InitWorkerNewThread(self.scheduler))
```

Comment:
you're not closing the client here:

```suggestion
        async def func(self):
            try:
                worker = get_worker()
            except ValueError:
                async with Client(self.scheduler, asynchronous=True) as client:
                    await client.register_worker_plugin(InitWorkerNewThread(self.scheduler))
```